en/test_and_ci/continous_integration.md

```diff
@@ -1,3 +1,100 @@
+The configuration files for these systems can be found in the [PX4-Autopilot](https://github.com/PX4/PX4-Autopilot) repo main branch:
+- [Jenkinsfile](https://github.com/PX4/PX4-Autopilot/blob/main/Jenkinsfile)
+- [.github/workflows/](https://github.com/PX4/PX4-Autopilot/tree/main/.github/workflows).
+
+The Jenkins build status can be viewed in [ci.px4.io:8080/job/PX4/](http://ci.px4.io:8080/job/PX4/)
+
+This document outlines how the different parts of the CI system are configured and connected.
+
+## Pull Requests
+
+Automated CI checks are run for every [Pull-Request (PR)](https://github.com/PX4/PX4-Autopilot/pulls).
+
+Most of the test are run using the Github workflows in [.github/workflows](https://github.com/PX4/PX4-Autopilot/tree/main/.github/workflows):
+
+- `compile_nuttx.yml`:
+  Builds NuttX flight controller board targets.
+  It can detect Flash overflows and code syntax errors,
+- `sitl_tests.yml`: 
+  Executes [MAVSDK SITL Tests](./integration_testing_mavsdk.md).
+  Detect system-level bugs in the code (e.g. VTOL doesn't return to home)
+- `mavros_mission_tests.yml`:
+  Executes mission plans using MAVROS.
+  Detects system level bugs in close, and also verifies compatibility between MAVROS and PX4.
+- `checks.yml`:
+  Runs various make targets (using `make` command).
+  The commands for each targets are documented in the [Makefile](https://github.com/PX4/PX4-Autopilot/blob/main/Makefile)
+  - `check_format`: Checks the coding style format using Clang Tidy ?
+  - `airframe_metadata`: Builds airframe XML metadata
+  - `parameters_metadata`: Builds parameter XML metadata
+  - `module_documentation`: Builds module documentation
+  - `tests`: Runs unit tests
+  - `tests_coverage`: Runs coverage tests
+  - ... (other auxiliary tests)
+- `ekf_functional_change_indicator.yml`: Check if there's any functionality change in EKF ?
+- `ekf_update_change_indicator.yml`: Check if there's any functionality change in EKF ?
+- `failsafe_sim.yml`: Builds [failsafe simulator](../config/safety_simulation.md)
+- `clang-tidy.yml`: Runs Clang Tidy to check if coding style is consistent
+- `compile_linux.yml`: Builds linux flight controller board targets
+
+## PX4 Releases/main
+
+A number of additional artifacts are built when changes are made to the `main` branch or to `release/*` branches (any releases).
+These are then deployed to other repositories, uploaded to servers, and generally made available to other systems and services that use them.
+
+Release/main actions are managed by the [.github/workflows/deploy_all.yml](https://github.com/PX4/PX4-Autopilot/blob/main/.github/workflows/deploy_all.yml) workflow.
+
+For each board target, the artifacts built are:
+
+- Parameter metadata (`parameters.*`)
+- Events metadata (`events/*.xz`)
+- Actuators metadata (`actuators.json*`)
+- Target binary (`*.px4`)
+
+These files are uploaded to the [Amazon S3 bucket server](https://px4-travis.s3.amazonaws.com/).
```

Comment:
Note, part where QGC fetches .px4 files from the S3:

https://github.com/mavlink/qgroundcontrol/blob/f9f04ecbbdabb7d2a160f7ab9152fbe45ef4d5cb/src/VehicleSetup/FirmwareUpgradeController.cc#L356-L362

Reply:
Example link: https://px4-travis.s3.amazonaws.com/Firmware/stable/px4_fmu-v5x_default.px4

Reply:
Also, the mechanism how QGC determines which version is stable, and beta is via this:

https://github.com/mavlink/qgroundcontrol/blob/f9f04ecbbdabb7d2a160f7ab9152fbe45ef4d5cb/src/VehicleSetup/FirmwareUpgradeController.cc#L654-L671

Reply:
Basically thru this link: https://api.github.com/repos/PX4/Firmware/releases

It returns a JSON file that includes the versioning info